docs/sphinx/releases/whats_new_1_5_0.rst

```diff
@@ -8,17 +9,560 @@
+to the C++ standard. All non-distributed modules are now in place, along with an
+experimental option to completely disable distributed features in HPX. We have
+also added an experimental asynchronous MPI executor. Lastly this release
+contains official |cmake| targets, performance improvements, and many bug fixes.
+
+* We have added the C++20 features ``hpx::jthread`` and ``hpx::stop_token``. We
+  have also added ``hpx::stable_sort`` based on Francisco Tapia's
+  implementation.
+* We have adapted existing sychronization primitives to be fully conformant
+  C++20: ``barrier``, ``latch``, ``counting_semaphore``.
+* We have started using customization point objects to make the corresponding
+  algorithms fully conformant to C++20 as well as to make algorithm extension
+  easier for the user. ``reduce``, ``copy``, ``all_of``/``any_of``/``none_of``,
+  ``fill``, ``count``, ``destroy``, ``move`` are using those CPOs.  We also have
+  adapted their corresponding ``hpx::ranges`` versions in this release,
+  deprecating the ``hpx::parallel`` ones.
+* In the perspective of continuing our work on making the hpx algorithms support
+  the ranges, we have adapted ``hpx::traits::is_range`` and others to use
+  ``hpx::traits::is_sentinel_for`` concept. We have adapted ``hpx::copy`` and
+  ``hpx::stable_sort`` algorithms.
+* We have adapted support for ``co_await`` to C++20, it now supports
+  ``shared_future`` and have added allocator support for futures returned by
+  ``co_return``. It is no longer in the ``experimental`` namespace.
+* We added serialization support for ``std::variant`` and ``std::tuple``.
+* ``result_of`` and ``is_callable`` are now deprecated and replaced by
+  ``invoke_result`` and ``is_invocable`` to conform to C++20.
+* We continued with the modularization, it is now easier to enable/disable the
+  distributed runtime option. An important amount of headers have been
+  deprecated. We adapted the namespaces and headers we could to be closer to the
+  standard ones (:ref:`api <_public_api>`).
+* It is now possible to have a basic CUDA support including a helper function to
+  get a future from a CUDA stream and target handling. They are available under
+  the ``hpx::cuda::experimental`` namespace and they can be enabled with the
+  ``HPX_WITH_ASYNC_CUDA=ON`` cmake option.
+* We added a new ``hpx::mpi::experimental`` namespace for getting futures from
+  an asynchronous MPI call and a new minimal MPI executor (add the name).
+* A polymorphic executor has been implemented to reduce compile times as a
+  function accepting executor can potentially be instanciated only once instead
+  of multiple times with different executors. It accepts the function signature
+  as a template argument. It needs to be constructed from any other executor.
+  Please note, that the function signatures that can be scheduled using
+  ``then_execute``, ``bulk_sync_execute``, ``bulk_async_execute`` and
+  ``bulk_then_execute`` are slightly different.
```

Comment:
Slightly different how?

Reply:
Should I detail the signatures here or refer to the corresponding [comment](https://github.com/STEllAR-GROUP/hpx/pull/4514#issuecomment-613139299) on Github?

Reply:
A link to the github ticket is probably fine.